src/internal/cluster/data.go

```diff
@@ -111,3 +116,24 @@ iterator:
 		return
 	}
 }
+
+// Pod filter to ensure we only use the current deployment's pods
+func podFilterByInitContainer(data types.ZarfDataInjection) k8s.PodFilter {
+	return func(pod corev1.Pod) bool {
+		// Loop over all init containers
+		for _, container := range pod.Spec.InitContainers {
```

Comment:
What if someone is using the data injections without the init-container pattern (some apps that dynamically pickup things from disk won't need to have an init container for inject)

Reply:
IMO we would want to look at the init containers in one loop and the real containers in a second loop (potentially breaking the insides out into another function).  We could also splice the arrays together potentially since they likely are the same struct type.